internal/service/mocks/collector.go

```diff
@@ -0,0 +1,77 @@
+// Code generated by mockery v2.12.2. DO NOT EDIT.
```

Comment:
This Collector isn't part of the service package. Seems like it should be in its own `mocks` directory under the `collector` package. I use this in opamp so having it closer to the interface it implements seems more intuitive for anyone wanting to use it.